webpack/components/TypeAhead/TypeAhead.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+
+import TypeAheadSearch from './pf3Search/TypeAheadSearch';
+// eslint-disable-next-line import/no-named-default
+import { default as TypeAheadSearchPf4 } from './pf4Search/TypeAheadSearch';
+import { getActiveItems } from './helpers/helpers';
+
+import './TypeAhead.scss';
+
+class TypeAhead extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: this.props.initialInputValue,
+    };
+
+    this.handleStateChange = this.handleStateChange.bind(this);
+  }
+
+  handleStateChange({ inputValue }) {
+    if (typeof inputValue === 'string') {
+      this.props.onInputUpdate(inputValue);
+      this.setState({ inputValue });
+    }
+  }
+
+  clearSearch = () => {
+    this.setState({ inputValue: '' }, () => this.props.onSearch(this.state.inputValue));
+  };
+
+  render() {
+    const {
+      onSearch, onInputUpdate, items, actionText, patternfly4, ...rest
+    } = this.props;
+
+    const activeItems = getActiveItems(items);
+
+    return (
+      <Downshift
+        onStateChange={this.handleStateChange}
+        defaultHighlightedIndex={0}
+        selectedItem={this.state.inputValue}
+        {...rest}
+      >
+        {({
+          getInputProps,
+          getItemProps,
+          isOpen,
+          inputValue,
+          highlightedIndex,
+          selectedItem,
+          selectItem,
+          openMenu,
+        }) => {
```

Comment:
I prefer to avoid render props when possible-- just a personal preference.  Seems like they have an alternate hooks API called `useCombobox`.. is it possible that could meet our needs here?

Trade you for a react-testing-library test :yum:

Reply:
I gave it a try and it's a bit of a can of worms because it requires refactoring this component into a functional component, which is proving to be difficult. 

Are there any specific concerns with using the render props? It is straight from the downshift documentation, I only moved it because I upgraded downshift and they deprecated the previous way of  passing them.

Reply:
The inversion of control makes it difficult to tell where things are coming from.  I can't tell at a glance where the passed-in values are coming from, and it requires a large cognitive load and poking around two or three files to figure it out.  I suppose it's not much worse than Redux, but at least that's the devil I know.

Reply:
Also it puts huge functions inside your JSX.